test-app/tests/integration/components/changeset-test.gts

```diff
@@ -0,0 +1,222 @@
+/* eslint-disable no-undef -- Until https://github.com/ember-cli/eslint-plugin-ember/issues/1747 is resolved... */
+/* eslint-disable simple-import-sort/imports,padding-line-between-statements,decorator-position/decorator-position -- Can't fix these manually, without --fix working in .gts */
+
+import { click, fillIn, render, setupOnerror } from '@ember/test-helpers';
+import { module, test, skip } from 'qunit';
+
+import HeadlessForm from 'ember-headless-form/components/headless-form';
+import validateChangeset from '@ember-headless-form/changeset/helpers/validate-changeset';
+import sinon from 'sinon';
+import { setupRenderingTest } from 'test-app/tests/helpers';
+import { Changeset } from 'ember-changeset';
+
+import type { ValidatorAction } from 'ember-changeset/types';
+
+module('Integration Component HeadlessForm > Changeset', function (hooks) {
+  setupRenderingTest(hooks);
+
+  interface TestFormData {
+    firstName?: string;
+    lastName?: string;
+  }
+
+  const validator: ValidatorAction = ({ key, newValue }) => {
+    const errors: string[] = [];
+
+    if (newValue == undefined) {
+      errors.push(`${key} is required!`);
+    } else if (typeof newValue !== 'string') {
+      errors.push('Unexpected type');
+    } else {
+      if (newValue.charAt(0).toUpperCase() !== newValue.charAt(0)) {
+        errors.push(`${key} must be upper case!`);
+      }
+
+      if (newValue.toLowerCase() === 'foo') {
+        errors.push(`Foo is an invalid ${key}!`);
+      }
+    }
+
+    return errors.length > 0 ? errors : true;
+  };
+
+  // Could not get this test to work, setupOnerror does not catch the error as expected here
```

Comment:
🤔 it should 🤔 that's goofy.

Reply:
I have seen errors in tests that could only be catched by plugging into qunit's own error handling IIRC... :think: